streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -484,7 +484,7 @@ public String queryableStoreName() {
         Objects.requireNonNull(named, "named can't be null");
 
         final String name = new NamedInternal(named).orElseGenerateWithPrefix(builder, TOSTREAM_NAME);
-        final ProcessorSupplier<K, Change<V>> kStreamMapValues = new KStreamMapValues<>((key, change) -> change.newValue);
+        final KStreamMapValues<K, Change<V>, V> kStreamMapValues = new KStreamMapValues<>((key, change) -> change.newValue);
         final ProcessorParameters<K, V, ?, ?> processorParameters = unsafeCastProcessorParametersToCompletelyDifferentType(
```

Comment:
Just FYI, @jeqo , during my POC, I tried to fix this right away, and it dragged me into migrating the entire DSL at once. I think we should do what you are doing instead: just migrate the individual processors first, and _then_ come back and drop the unsafe casts in a later PR.